bcl2fastq/src/main/java/com/hartwig/bcl2fastq/conversion/ResultAggregation.java

```diff
@@ -95,7 +95,15 @@ static ConvertedFastq fastq(IlluminaSample sample, Map.Entry<String, List<Blob>>
     }
 
     private static String outputPath(final IlluminaSample sample, final Stats stats, final Blob blobR1) {
-        return stats.flowcell() + "/" + sample.barcode() + "/" + new File(blobR1.getName()).getName();
+        String[] fileNameSplit = new File(blobR1.getName()).getName().split("_");
```

Comment:
We have had surprises in the past, would it be wise to check the filename conforms to expectations?

Reply:
hmm, as this file convention is produced by the version of bcl2fastq also in this codebase I think its fine to have this implicit dependency on the filename.